coral/src/app/features/connectors/details/ConnectorDetails.tsx

```diff
@@ -41,27 +46,67 @@ function ConnectorDetails(props: ConnectorOverviewProps) {
   const { connectorName } = props;
   const { state: initialEnvironment }: { state: string | null } = useLocation();
 
-  const matches = useMatches();
-  const currentTab = findMatchingTab(matches);
-
+  const [showClaimModal, setShowClaimModal] = useState(false);
   const [environmentId, setEnvironmentId] = useState<string | undefined>(
     initialEnvironment ?? undefined
   );
 
+  const [claimErrorMessage, setClaimErrorMessage] = useState<
+    string | undefined
+  >();
+
+  const toast = useToast();
+
+  const matches = useMatches();
+
+  const currentTab = findMatchingTab(matches);
+
   const {
     data: connectorData,
     isError: connectorIsError,
     error: connectorError,
     isLoading: connectorIsLoading,
     isRefetching: connectorIsRefetching,
-  } = useQuery(["connector-overview", connectorName, environmentId], {
+    refetch: refetchConnectors,
+  } = useQuery({
+    queryKey: ["connector-overview", connectorName, environmentId],
     queryFn: () =>
       getConnectorOverview({
         connectornamesearch: connectorName,
         environmentId,
       }),
   });
 
+  const {
+    mutate: createClaimConnectorRequest,
+    isLoading: isLoadingCreateClaimConnectorRequest,
+    isError: isErrorCreateClaimConnectorRequest,
+  } = useMutation(
+    (remark?: string) =>
+      requestConnectorClaim({
+        connectorName: connectorData?.connectorInfo.connectorName || "",
```

Comment:
Should we actually throw an error when `connectorName` or `environmentId` is `undefined`? I know it's probably safe to have it like this, but it's still a little uncomfortable ^^

Reply:
The Banner is not rendered when `connectorData.connectorInfo` is not defined, so this the mutation can never be called with an undefined environmentId or connectorName. 

I would not add error handling really in that case - BUT I think it would be a good idea to add a `console.error` for developers. That way, in case someone would change something that could break that. (bc devs are the only ones who can break that), they would see it. And it's also would give more context for reading the code. 

WDYT?

Reply:
Yes, should have clarified that no action was really required. But your solution is good, thank you ^^